components/ArticleList.jsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from "react";
+import { articleList } from "../css/ArticleList.module.css";
+import ArticleCard from "./ArticleCard.jsx";
+import { loading, error } from "../utils/htmlUtils";
+import { getArticles } from "../utils/api";
+
+const ArticleList = () => {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  useEffect((params) => {
+    getArticles(setArticles)
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
```

Comment:
I'm not convident this will catch erros properly, instead of passing setArticles into getArticles, you should have a then block which you call setArticles from.